lib/zooniverse_user.rb

```diff
@@ -1,4 +1,77 @@
 class ZooniverseUser < ActiveRecord::Base
   establish_connection :"zooniverse_home_#{ Rails.env }"
   self.table_name = 'users'
+
+  attr_reader :password
+
+  def self.import_users(users=nil)
+    if users.blank?
+      self
+    else
+      where(login: users)
+    end.find_each(&:import)
+  end
+
+  def self.authenticate(username, password)
+    find_by(login: username).try(:authenticate, password)
+  end
+
+  def self.from_user(user)
+    zu = find_or_create_by(login: user.display_name) do |u|
+      u.name = user.display_name
+      u.password = user.password
+      u.email = user.email
+      u.set_tokens!
+    end
+    zu.save!
+    user.update!(zooniverse_id: zu.id.to_s)
+    zu
+  end
+
+  def password=(plain_password)
+    return unless plain_password
+    @password = plain_password
+    self.password_salt = simple_token
+    self.crypted_password = Sha1Encryption.encrypt(plain_password, salt: password_salt)
+  end
+
+  def authenticate(plain_password)
+    crypted_password == Sha1Encryption.encrypt(plain_password, salt: password_salt) ? self : nil
+  end
+
+  def import
+    user = User.find_or_initialize_by(zooniverse_id: id.to_s) do |u|
+      u.display_name = login
```

Comment:
I can't I need it to exist to build the identity group. I can rewrite so the fields aren't set in the block and instead it can test if the resource already exists and then assign those fields.